crates/subspace-farmer-components/src/piece_caching.rs

```diff
@@ -0,0 +1,50 @@
+use lru::LruCache;
+use parking_lot::Mutex;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use subspace_core_primitives::{Piece, PieceIndexHash};
+use tracing::trace;
+
+const CACHE_ITEMS_LIMIT: usize = 10000; // TODO: adjust after piece size change
+
+#[derive(Clone)]
+pub struct PieceMemoryCache {
+    cache: Arc<Mutex<LruCache<PieceIndexHash, Piece>>>,
+}
+impl Default for PieceMemoryCache {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PieceMemoryCache {
+    pub fn new() -> Self {
+        Self {
+            cache: Arc::new(Mutex::new(LruCache::new(
+                NonZeroUsize::new(CACHE_ITEMS_LIMIT).expect("Manually set value > 0."),
+            ))),
+        }
```

Comment:
What about making this "fn default()", but still allowing customizing size through `fn new()` if desired? It feels odd to not be able to customize it. Or default can be `1` meaning no caching (might be useful for tests).